File: src/Player-Stage-Manual-4.1.0/code/Ch6.3/bigbob8.cc
/* bigbob8.cc - demo GetSetSpeed and GetPositions
 * K. Nickels 7/24/13
 */

#include <iostream>
#include <libplayerc++/playerc++.h>

int main(int argc, char *argv[]) {
  using namespace PlayerCc;

  PlayerClient    robot("localhost");
  Position2dProxy p2dProxy(&robot,0);

  // read from the proxies
  robot.Read();

  // for carlike robots
  // p2dProxy.SetCarlike(0.2,dtor(3)); // XSpeed, Yaw (rad)
  // for holonomic robots
  // p2dProxy.SetSpeed(0.1,0.1,0.1); // XSpeed (m/s), YSpeed, YawSpeed (rad/s)
  // for nonholonomic robots
  p2dProxy.SetSpeed(0.1,0,0.1); // XSpeed (m/s), YawSpeed (rad/s)

  for(int i=0;i<10;i++) {
        robot.Read();
        std::cout << "Iter "<<i<<std::endl;
        std::cout << "XSpeed = " << p2dProxy.GetXSpeed() << " m/s, " <<
                "YSpeed = " << p2dProxy.GetYSpeed() << " m/s, " <<
                "YawSpeed = " << p2dProxy.GetYawSpeed() << " rad/s"<< std::endl;
        std::cout << "XPos = " << p2dProxy.GetXPos() << " m, " <<
                "YPos = " << p2dProxy.GetYPos() << " m, " <<
                "Yaw = " << p2dProxy.GetYaw() << " rad"<< std::endl;
    }

    return 0;
}

File: src/Player-Stage-Manual-4.1.0/code/Ch8.3/bigbob8.py
# bigbob8.py - demo GetSpeed GetPos
# Original author: K. Nickels 7/24/13

import sys, os, math
# This should be wherever "playercpp.py" is
sys.path.append('/usr/local/lib/python2.7/site-packages/')
sys.path.append('/usr/local/lib64/python2.7/site-packages/')
from playercpp import *


# read from the proxies
robot = PlayerClient("localhost");
pp = Position2dProxy(robot,0);


# for carlike robots
# p2dProxy.SetSpeed(0.1,0.1); # XSpeed (m/s), YawSpeed (rad/s)
pp.SetCarlike(0.2,math.radians(3.0)); # XSpeed, Yaw (rad)
# for holonomic robots
# pp.SetSpeed(0.1,0.1,0.1); # XSpeed (m/s), YawSpeed (rad/s)

for i in range(10):
	robot.Read()
	print 'Iter ',i
	print 'XSpeed = %.3f m/s, ' % pp.GetXSpeed(), 
	print 'YSpeed = %.3f m/s, ' % pp.GetYSpeed(), 
	print 'YawSpeed = %.3f rad/s ' % pp.GetYawSpeed()
	print 'XPos = %.3f m, ' % pp.GetXPos(),
	print 'YPos = %.3f m, ' % pp.GetYPos(),
	print 'Yaw = %.3f rad' % pp.GetYaw() 

# Destructors called automatically on exit.
